frontend/src/lib/components/AnnotationsOverlay/useAnnotationsPositioning.ts

```diff
@@ -0,0 +1,43 @@
+import { Chart } from 'chart.js'
+import { useMemo } from 'react'
+
+export interface AnnotationsPositioning {
+    tickIntervalPx: number
+    firstTickLeftPx: number
+    pointsPerTick: number // TODO: Use this
```

Comment:
TODO?

Reply:
This is some extra complexity to handle an edge case with dense graphs, so I'll address it in a separate PR

Reply:
Removed for now though as the code that's in there currently is simple anyway